### PR TITLE
first pull request, bidness logic and DrawButton

### PR DESCRIPTION
**Pull request by MaxwellChalmers**
# What is included in this change?

I got the rules of the card game (only draw up to 3 new cards, 4 cards with an ace that stays in hand) and also the new game button working.  
I added a DrawButton.js file to deal with the button better. 

# Describe at least one problem you solved

I had to figure out how to properly update the variable I used to track the number of aces after running the new game version of fetchNewCards. _I definitely wasn't just passing the newHand directly into setNumOfAces without calling my helper function_ TT

also how to make sure the newGame variable is updated before running fetchNewCards.

# How did you test this change?

I tested my changes with docker and Firefox mostly, can't say every possible edge case is proven to work but I would expect it to have consistent performance at this point.

# If things are not currently working, explain.

everything changed seems to be working fine currently

# UI Changes 

insignificant visually, sometimes the button says "new game?" instead of "Draw!"

### CODE REVIEW

File: src/ui/DrawButton.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { useState, useCallback } from "react";
```
Comment (semmypurewal) [inline src/ui/DrawButton.js:1]: I don't think you're using `useCallback` here, so you can remove it.

File: src/ui/DrawButton.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+export default function DrawButton({ onClick }) {
+  const [HasDrawn, setHasDrawn] = useState(false);
+
+  const checkClick = async () => {
+    setHasDrawn(!HasDrawn);
+    await onClick();
+  };
+
+  return (
+    <button onClick={checkClick}>{HasDrawn ? "New Game?" : "Draw!"}</button>
```
Comment (semmypurewal) [inline src/ui/DrawButton.js:12]: This is good. Do you know Nathaniel? He was asking how to do if expressions in React yesterday. Would be cool if you would share this example with him.

File: src/ui/App.js [hunk 3]
```diff
@@ -11,17 +11,53 @@ import { useState, useCallback } from "react";
...
+      if (isAce) {
+        setUnselectedAcesCount((prevCount) => prevCount + 1);
+      }
     }
   }
 
   // This function will be called when the Draw button is clicked
   const fetchNewCards = useCallback(async () => {
+    let s = selected.length;
+    if (newGame) {
```
Comment (semmypurewal) [inline src/ui/App.js:57]: This is smart! But keep in mind if you play too many games you'll exhaust your deck. :)

**Review by semmypurewal — COMMENTED** [review]
This is great. I can tell you've though through a lot of things and that you've hacked things in place so that they work. That's awesome.

Let me know if you need any other feedback on it, but it's definitely fine for now. Great work!
